com.unity.render-pipelines.high-definition/Runtime/ShaderLibrary/ShaderGraphFunctions.hlsl

```diff
@@ -27,6 +27,19 @@ float3 shadergraph_HDBakedGI(float3 positionWS, float3 normalWS, float2 uvStatic
     return SampleBakedGI(positionRWS, normalWS, uvStaticLightmap, uvDynamicLightmap);
 }
 
+void shadergraph_HDGetSunLight(out float3 direction, out float3 color)
+{
+    direction = 0.0f;
+    color = 0.0f;
+#if defined(SHADERPASS) && (SHADERPASS != SHADERPASS_LIGHT_TRANSPORT)
+    if (_DirectionalLightCount > 0)
+    {
+        direction = -_DirectionalLightDatas[0].forward;
```

Comment:
I am not sure this work currently. Is the main line always the slot 0 when we have multiple directional light? Not sure about that.

Reply:
indeed, when the sun light has shadows, it may not always be the first one, so i used _DirectionalShadowIndex in this case